pontoon/base/migrations/0048_translationmemoryentry.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+import django.db.models.deletion
+
+from django.contrib.postgres.operations import CreateExtension
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('base', '0047_repository_permalink_prefix'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='TranslationMemoryEntry',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('source', models.TextField()),
+                ('target', models.TextField()),
+                ('entity', models.ForeignKey(on_delete=django.db.models.deletion.SET_NULL, to='base.Entity', null=True)),
+                ('locale', models.ForeignKey(to='base.Locale')),
+                ('translation', models.ForeignKey(related_name='memory_entries', on_delete=django.db.models.deletion.SET_NULL, to='base.Translation', null=True)),
+            ],
+        ),
+        CreateExtension('fuzzystrmatch'),
+        migrations.RunSQL('CREATE INDEX source_length_idx ON base_translationmemoryentry(CHAR_LENGTH(source))',
+                          'DROP INDEX source_length_idx'),
+        # Equivalent function to the Levenshtein.ratio()
+        migrations.RunSQL('''CREATE OR REPLACE FUNCTION levenshtein_ratio(string1 text, string2 text) RETURNS float AS $$
```

Comment:
Stored procedures in the database are generally undesirable because there's no singular place where the code for them is stored. In the future, if we modify this, we have to add another migration rather than edit some existing code, and it won't be obvious from reading where the procedure is used where the code for it lives.

I'd suggest instead moving this into the `extra` call explicitly, and then moving that `extra` call into a queryset method for the translation memory, so you end up just calling something like `qs.minimum_levenshtein_ratio(text, min_quality)` or whatever.